colobot-base/src/common/system/system.h

```diff
@@ -75,14 +75,19 @@ enum class SystemDialogResult
 class CSystemUtils
 {
 public:
+    CSystemUtils();
```

Comment:
This constructor should be protectedc instead of public

Reply:
Making the constructor protected is unnecessary because the class is already abstract, it contains pure virtual methods and cannot be instantiated. However, I made the destructor pure virtual to make this more clear.

Reply:
More clear? Nope. The opposite.
1. Someone who missed that the class has a pure virtual method (e.g. me) will just as easily miss a virtual destructor.
2. Someone who does notice the destructor will now have more questions. The first question will be "does this work?" (b.t.w [it does not unless you implement the destructor](https://stackoverflow.com/a/630960/14516046)) and the second question will be "why does this class need a pure virtual destructor?"